organization/ircam_oauth2_provider/views.py

```diff
@@ -0,0 +1,114 @@
+    authorize_url = '{}/o/authorize/'.format(settings.USER_SERVER_BASEURL)  # Accessed by the client so must be host-reachable
+    profile_url = '{}/profile/'.format(settings.OAUTH_SERVER_BASEURL)
+
+    def create_or_update_person(self,user):
+        try :
+            person, created = Person.objects.get_or_create(user_id=user.id)            
+            person.user = user
+            person.title = user.first_name+" "+user.last_name
+            person.first_name = user.first_name
+            person.last_name = user.last_name
+            person.email = user.email
+
+            person.save()
+
+        except ObjectDoesNotExist:
+            person = Person.objects.get(
+                user_id=user.id
+            )
+        except IntegrityError:
+            pass
+
+    def create_user_socialaccount(self,request,extra_data):
+
+        user, created = User.objects.update_or_create(
+            username = extra_data['username'],
+            defaults={
+                'first_name': extra_data['first_name'],
+                'last_name': extra_data['last_name'],
+                'is_active': True,
+                'is_superuser': False,
+                'is_staff': False,
+                'email': extra_data['email'],
+            }
+        )
+        user.save()
+        self.logger.info('Local user {0} id:{1} username:{2} email:{3}'.format('created: ' if created else 'updated: ',user.id,user.username,user.email))
+        
+        # Creating or updating Person
+        self.create_or_update_person(user)
+       
+        # Creating the allauth social account so the user can log in through Ircam Auth
+        social_account = SocialAccount(user=user,
+                                    provider='ircamauth',
+                                    uid=extra_data['id'],
+                                    extra_data={'id': user.id, 'username': user.username, 'email': user.email})
+        social_account.save()
+        self.logger.info('Social account created. User {0} '.format(user.username))
+        return self.get_provider().sociallogin_from_response(request, extra_data)
+
+    def update_localuser(self,social_user,extra_data):
+        from django.contrib.auth.models import User
+
+        user = User.objects.filter( username__contains=extra_data['username'] )[0]
+        if (  
+                user.email != extra_data['email'] or 
+                user.first_name != extra_data['first_name'] or 
+                user.last_name != extra_data['last_name'] 
+            ) :
+            user.username = extra_data['username']
+            user.email = extra_data['email']
+            user.first_name = extra_data['first_name']
+            user.last_name = extra_data['last_name']
+            user.is_active = True
+            user.save()
+            self.logger.info('User {0} updated, email:{1}, firstN:{2}, LastN:{3}'
+                .format(user.username,user.email,user.first_name,user.last_name))
```

Comment:
It leaks user data, it should be a debug log.

Reply:
The idea was to trace user creation and connection in production. If we don't need it in production let me know.